src/approaches/open_loop_llm_approach.py

```diff
@@ -75,40 +100,22 @@ def _predict(self, state: State, atoms: Set[GroundAtom],
             temperature=CFG.open_loop_llm_temperature,
             seed=CFG.seed,
             num_completions=CFG.open_loop_llm_num_completions)
-        # Try to convert the output into an abstract plan.
-        for llm_prediction in llm_predictions:
-            ground_nsrt_plan = self._process_single_prediction(
-                llm_prediction, state, atoms, goal)
-            if ground_nsrt_plan is not None:
-                # If valid plan, add plan to memory so it can be refined!
-                memory["abstract_plan"] = ground_nsrt_plan
-                return memory["abstract_plan"].pop(0)
-        # Give up if none of the predictions work out.
-        raise ApproachFailure("No LLM predicted plan achieves the goal.")
+        for pred in llm_predictions:
+            option_plan = self._llm_prediction_to_option_plan(pred, objects)
+            yield option_plan
 
-    def _process_single_prediction(
-            self, llm_prediction: str, state: State, atoms: Set[GroundAtom],
+    def _option_plan_to_nsrt_plan(
+            self, option_plan: List[Tuple[ParameterizedOption,
+                                          Sequence[Object]]],
+            atoms: Set[GroundAtom], objects: Set[Object],
             goal: Set[GroundAtom]) -> Optional[List[_GroundNSRT]]:
-        objects = set(state)
-        option_plan = self._llm_prediction_to_option_plan(
-            llm_prediction, objects)
-        # If we failed to find a nontrivial plan with this prediction,
-        # continue on to next prediction.
-        if len(option_plan) == 0:
```

Comment:
where did this logic go? did it turn out to be automatically handled by the existing logic in `task_plan_with_option_plan_constraint`?

Reply:
yeah, i'm not sure why we had it in the first place